---
layout: slim_banner
title: EISA - Exon/Intron-split Analysis
---

## Reference
This page contains links to supplementary online files for Exon-Intron Split analysis (EISA):

**Analysis of intronic and exonic reads in RNA-seq data characterizes transcriptional and post-transcriptional regulation.**  
Dimos Gaidatzis, Lukas Burger, Maria Florescu and Michael B. Stadler  
*Nature Biotechnology*, 2015 Jul;33(7):722-9. doi: 10.1038/nbt.3269. Epub 2015 Jun 22.  
[Abstract](http://www.nature.com/nbt/journal/v33/n7/abs/nbt.3269.html) |
[doi:10.1038/nbt.3269](http://doi.org/10.1038/nbt.3269) |
[Full text (HTML)](http://www.nature.com/nbt/journal/vaop/ncurrent/full/nbt.3269.html) |
[Full text (PDF)](http://www.nature.com/nbt/journal/v33/n7/pdf/nbt.3269.pdf)  

## Supplementary Online Files
(to download archives of these files, see the bottom of this page)

### List of non-overlapping genes for each genome assembly and strandedness
[H. sapiens hg18 stranded](nonOverlappingGenes_hg18_stranded.txt.gz)  
[H. sapiens hg18 unstranded](nonOverlappingGenes_hg18_unstranded.txt.gz)  
[M. musculus mm9 stranded](nonOverlappingGenes_mm9_stranded.txt.gz)  
[M. musculus mm9 unstranded](nonOverlappingGenes_mm9_unstranded.txt.gz)  

### Figure 2a
- Bhatt et al. 2012  
  [Fig2a_GSE32916_rawcounts_exonic.txt.gz](Fig2a_GSE32916_rawcounts_exonic.txt.gz) (0.4MB)  
  [Fig2a_GSE38892_rawcounts_exonic.txt.gz](Fig2a_GSE38892_rawcounts_exonic.txt.gz) (0.2MB)  
  [Fig2a_GSE38892_rawcounts_intronic.txt.gz](Fig2a_GSE38892_rawcounts_intronic.txt.gz) (0.2MB)  

- Gene clusters from Bhatt et al. 2012 and expression changes as compared to the first time point  
  [Fig2a_genesInBhatt2012Clusters_normalized.txt.gz](Fig2a_genesInBhatt2012Clusters_normalized.txt.gz) (0.1MB)  

### Figure 2bcd
- Menet et al. 2012  
  [Fig2bcd_GSE36916_rawcounts_exonic.txt.gz](Fig2bcd_GSE36916_rawcounts_exonic.txt.gz) (0.6MB)  
  [Fig2bcd_GSE36916_rawcounts_intronic.txt.gz](Fig2bcd_GSE36916_rawcounts_intronic.txt.gz) (0.6MB)  
  [Fig2bcd_GSE36916_normalized_forPhaseFitting.txt.gz](Fig2bcd_GSE36916_normalized_forPhaseFitting.txt.gz) (1.1MB)  

- Koike et al. 2012  
  [Fig2bcd_GSE39978_rawcounts_exonic.txt.gz](Fig2bcd_GSE39978_rawcounts_exonic.txt.gz) (0.3MB)  
  [Fig2bcd_GSE39978_rawcounts_intronic.txt.gz](Fig2bcd_GSE39978_rawcounts_intronic.txt.gz) (0.3MB)  
  [Fig2bcd_GSE39978_normalized_forPhaseFitting.txt.gz](Fig2bcd_GSE39978_normalized_forPhaseFitting.txt.gz) (1.3MB)  

- Vollmers et al. 2012  
  [Fig2bcd_SRA025656_rawcounts_exonic.txt.gz](Fig2bcd_SRA025656_rawcounts_exonic.txt.gz) (0.3MB)  
  [Fig2bcd_SRA025656_rawcounts_intronic.txt.gz](Fig2bcd_SRA025656_rawcounts_intronic.txt.gz) (0.3MB)  
  [Fig2bcd_SRA025656_normalized_forPhaseFitting.txt.gz](Fig2bcd_SRA025656_normalized_forPhaseFitting.txt.gz) (0.7MB)  

- Fitted phases and amplitudes for detected genes (all three datasets)  
  [Fig2bcd_fittedPhasesAmplitudes_detectedGenes.txt.gz](Fig2bcd_fittedPhasesAmplitudes_detectedGenes.txt.gz) (0.6MB)  

### Figure 2ef
- Jin et al. 2013  
  [Fig2ef_GSE43070_rawcounts_exonic.txt.gz](Fig2ef_GSE43070_rawcounts_exonic.txt.gz) (0.2MB)  
  [Fig2ef_GSE43070_rawcounts_intronic.txt.gz](Fig2ef_GSE43070_rawcounts_intronic.txt.gz) (0.2MB)  
  [Fig2ef_GSE43070_rawcounts_groSeq.txt.gz](Fig2ef_GSE43070_rawcounts_groSeq.txt.gz") (0.2MB)  
  [Fig2ef_GSE43070_normalized_exonic_intronic_groSeq.txt.gz](Fig2ef_GSE43070_normalized_exonic_intronic_groSeq.txt.gz) (0.5MB)  

### Figure 3abc
- Tippmann et al. 2012  
  [Fig3abc_GSE33252_rawcounts_exonic.txt.gz](Fig3abc_GSE33252_rawcounts_exonic.txt.gz) (0.2MB)  
  [Fig3abc_GSE33252_rawcounts_intronic.txt.gz](Fig3abc_GSE33252_rawcounts_intronic.txt.gz) (0.2MB)  
  [Fig3abc_GSE33252_normalized_exonic_intronic.txt.gz](Fig3abc_GSE33252_normalized_exonic_intronic.txt.gz) (0.4MB)  
  [Fig3abc_GSE33252_normalized_exonic_intronic_halflives.txt.gz](Fig3abc_GSE33252_normalized_exonic_intronic_halflives.txt.gz) (0.2MB)  

### Figure 4ab
- Guo et al. 2010  
  [Fig4ab_GSE21992_rawcounts_exonic.txt.gz](Fig4ab_GSE21992_rawcounts_exonic.txt.gz) (0.2MB)  
  [Fig4ab_GSE21992_rawcounts_intronic.txt.gz](Fig4ab_GSE21992_rawcounts_intronic.txt.gz) (0.2MB)  
  [Fig4ab_GSE21992_normalized_exonic_intronic_mirSites.txt.gz](Fig4ab_GSE21992_normalized_exonic_intronic_mirSites.txt.gz) (0.6MB)  

### Figure 4cd
- Nam et al. 2014  
  [Fig4cd_GSE52530_rawcounts_exonic.txt.gz](Fig4cd_GSE52530_rawcounts_exonic.txt.gz) (0.6MB)  
  [Fig4cd_GSE52530_rawcounts_intronic.txt.gz](Fig4cd_GSE52530_rawcounts_intronic.txt.gz) (0.6MB)  
  [Fig4cd_GSE52530_Huh7_normalized_exonic_intronic_mirSites.txt.gz](Fig4cd_GSE52530_Huh7_normalized_exonic_intronic_mirSites.txt.gz) (0.7MB)  
  [Fig4cd_GSE52530_IMR90_normalized_exonic_intronic_mirSites.txt.gz](Fig4cd_GSE52530_IMR90_normalized_exonic_intronic_mirSites.txt.gz) (0.4MB)  
  [Fig4cd_GSE52530_HEK293_normalized_exonic_intronic_mirSites.txt.gz](Fig4cd_GSE52530_HEK293_normalized_exonic_intronic_mirSites.txt.gz) (0.8MB)  
  [Fig4cd_GSE52530_HeLa_normalized_exonic_intronic_mirSites.txt.gz](Fig4cd_GSE52530_HeLa_normalized_exonic_intronic_mirSites.txt.gz) (0.7MB)  

### Figure 5ab
- ENCODE cell lines  
  [Fig5ab_GSE30567_PolyA_rawcounts_exonic.txt.gz](Fig5ab_GSE30567_PolyA_rawcounts_exonic.txt.gz) (0.3MB)  
  [Fig5ab_GSE30567_PolyA_rawcounts_intronic.txt.gz](Fig5ab_GSE30567_PolyA_rawcounts_intronic.txt.gz) (0.3MB)  
  [Fig5ab_GSE30567_nonPolyA_rawcounts_exonic.txt.gz](Fig5ab_GSE30567_nonPolyA_rawcounts_exonic.txt.gz) (0.3MB)  
  [Fig5ab_GSE30567_nonPolyA_rawcounts_intronic.txt.gz](Fig5ab_GSE30567_nonPolyA_rawcounts_intronic.txt.gz) (0.3MB)  
  [Fig5ab_GSE30567_normalized_exonic_intronic.txt.gz](Fig5ab_GSE30567_normalized_exonic_intronic.txt.gz) (1.4MB)  

### Figure 5cde
- Illumina Human Body Map 2.0  
  [Fig5cde_GSE30611_rawcounts_exonic.txt.gz](Fig5cde_GSE30611_rawcounts_exonic.txt.gz) (0.6MB)  
  [Fig5cde_GSE30611_rawcounts_intronic.txt.gz](Fig5cde_GSE30611_rawcounts_intronic.txt.gz) (0.5MB)  
  [Fig5cde_GSE30611_normalized_exonic_intronic_mirSites.txt.gz](Fig5cde_GSE30611_normalized_exonic_intronic_mirSites.txt.gz) (3.2MB)  

### Code
- Supplementary Script 1: Example R code for fitting phases/amplitudes of oscillating genes (dataset from Figure 2)  
  [EISA_fitCircadianPhases_R-code.R](EISA_fitCircadianPhases_R-code.R)  
- Supplementary Script 2: Example R code using edgeR (dataset from Figure 3)  
  [EISA_edgeR_R-code.R](EISA_edgeR_R-code.R)  

## Archives with above files grouped by article figure
  [Supplementary_Data_1_nonOverlappingGenes.tar](Supplementary_Data_1_nonOverlappingGenes.tar) (lists of non-overlapping genes, 0.2MB)  
  [Supplementary_Data_2_dataFigure2.tar](Supplementary_Data_2_dataFigure2.tar) (data for Figure 2, 7.7MB)  
  [Supplementary_Data_3_dataFigure3.tar](Supplementary_Data_3_dataFigure3.tar) (data for Figure 3, 1.0MB)  
  [Supplementary_Data_4_dataFigure4.tar](Supplementary_Data_4_dataFigure4.tar) (data for Figure 4, 4.8MB)  
  [Supplementary_Data_5_dataFigure5.tar](Supplementary_Data_5_dataFigure5.tar) (data for Figure 5, 7.0MB)  
  [Supplementary_Scripts.tar](Supplementary_Scripts.tar) (Supplementary Scripts 1 and 2)  
